addons/website_event/models/event_event.py

```diff
@@ -55,11 +56,17 @@ def _compute_website_url(self):
             if event.id:  # avoid to perform a slug on a not yet saved record in case of an onchange.
                 event.website_url = '/event/%s' % slug(event)
 
-    @api.onchange('event_type_id')
-    def _onchange_type(self):
-        super(Event, self)._onchange_type()
-        if self.event_type_id:
-            self.website_menu = self.event_type_id.website_menu
+    @api.depends('event_type_id')
+    def _compute_from_event_type(self):
+        """ Also ensure a value for website_menu as it is a trigger notably for
+        track related menus. """
+        super(Event, self)._compute_from_event_type()
+        for event in self:
+            if not event.event_type_id:
+                if not event.website_menu:
+                    event.website_menu = False
+            elif event.event_type_id:
+                event.website_menu = event.event_type_id.website_menu
```

Comment:
Inverse for clarity?

```py
if event.event_type_id:
    event.website_menu = event.event_type_id.website_menu
elif not event.website_menu:
    event.website_menu = False
```

(Feels a bit strange now that I'm inverting it, is it to fill None values? Then I prefer a "is None" for clarity)

Reply:
Actually it is to keep the same heuristic than the base _compute_from_event_type method. Anyway I concatenated two "if". Concerning the "is None" I prefer to stay neutral, we can write False on False that's not an issue.